packages/config/package.json

```diff
@@ -5,8 +5,11 @@
   "description": "Tools to read configuration for @glint packages",
   "license": "MIT",
   "author": "Dan Freeman (https://github.com/dfreeman)",
-  "main": "lib/index.js",
-  "types": "lib/index.d.ts",
+  "type": "module",
+  "exports": {
+    ".": "./lib/index.js",
+    "./types": "./lib/types.cjs"
```

Comment:
TS doesn't allow you to import types from an ESM module into a CJS one, and the environment packages (which _have_ to be CJS) want to be able to reference types from `@glint/config` in their definitions to ensure they're adhering to the interface correctly.

To make that work, we have a single `.cts` file that just contains type declarations that can be imported by both ESM and CJS, and here we're setting it that up to be exposed to all comers, regardless of module flavor, as `@glint/config/types`.

Reply:
So that all makes sense; the thing I am confused about is that it is a `.cjs` file instead of a `.cts` file for types. (I should have been clearer in my first question!)

Reply:
The output on disk will be a `.cjs` file and a `.d.cts` one, so we're mapping to it the same way we would any other `.js`/`.d.ts` or `.mjs`/`.d.mts` pair for consistency.

Reply:
Huh. I did not realize the `types` key worked that way. TIL.

Reply:
That is, for posterity, *not* the `types` key. The confusion: `"types"` can be a key of the same sort as `"import"`, `"require"`, and `"default"`, but this is a *path* to the *file* which exports the relevant types, as indicated by the leading `./`.